### PR TITLE
Add resuming from specific checkpoint

### LINKED ISSUE #515: Add resuming from specific checkpoint
**Is your feature request related to a problem? Please describe.**
In current version, there is no way to resume training from a specific checkpoint (not the last checkpoint).
Sometimes (very often in my case), one needs experiment training with different hyperparameters (e.g. dropout rate, augmentation) from a specific checkpoint.

**Describe the solution you'd like**
Add `resume_from_checkpoint`argument to Trainer class.

**Describe alternatives you've considered**
I tried to use `restore` from Trainer class, but was not successful because it is meant to be used only when .fit() is called.

**Additional context**
FYI, I made a PR to demonstrate my idea: https://github.com/williamFalcon/pytorch-lightning/pull/516

### PR DESCRIPTION
**Pull request by dreamgonfly**
# Before submitting

- [ v ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ v ] Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [ v ] Did you make sure to update the docs?   
- [  ] Did you write any new necessary tests?

## What does this PR do?
Fixes #515

## Did you have fun?
I solved my problem with this feature.

### CODE REVIEW

File: pytorch_lightning/trainer/trainer_io.py [hunk 1]
```diff
@@ -93,6 +96,18 @@ def restore_state_if_checkpoint_exists(self, model):
 
         return did_restore
 
+    def restore_state_from_checkpoint(self, checkpoint_path):
+        did_restore = False
+
+        checkpoint_path = Path(checkpoint_path)
+        if not checkpoint_path.exists():
+            return did_restore
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer_io.py]: You can simply return True/False no need for extra variable

File: pytorch_lightning/trainer/trainer_io.py [hunk 2]
```diff
@@ -93,6 +96,18 @@ def restore_state_if_checkpoint_exists(self, model):
 
         return did_restore
 
+    def restore_state_from_checkpoint(self, checkpoint_path):
+        did_restore = False
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer_io.py]: add doc what data type the `checkpoint_path` is

File: pytorch_lightning/trainer/trainer_io.py [hunk 3]
```diff
@@ -93,6 +96,18 @@ def restore_state_if_checkpoint_exists(self, model):
...
+
+        checkpoint_path = Path(checkpoint_path)
+        if not checkpoint_path.exists():
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer_io.py]: does not work for `str` as it is defined `:param resume_from_checkpoint: str or os.PathLike object.`

File: pytorch_lightning/trainer/trainer_io.py (hunk 1 above)
Comment (dreamgonfly) [reply pytorch_lightning/trainer/trainer_io.py]: Okay. I'll change the code accordingly.

Comment (dreamgonfly) [reply pytorch_lightning/trainer/trainer_io.py]: I simplified the code and this function no longer exists

File: pytorch_lightning/trainer/trainer_io.py (hunk 2 above)
Comment (dreamgonfly) [reply pytorch_lightning/trainer/trainer_io.py]: I removed this function. Please review the updated code :)

File: pytorch_lightning/trainer/trainer_io.py (hunk 3 above)
Comment (dreamgonfly) [reply pytorch_lightning/trainer/trainer_io.py]: checkpoint_path is what torch.load expects as input. It can be file-like object or str containing a file name.
https://pytorch.org/docs/stable/torch.html?highlight=torch%20load#torch.load

**Comment by dreamgonfly** [comment]
I updated & simplified the code.

If both resume_from_checkpoint and last checkpoint in checkpoint_callback.filepath exist, Trainer restores checkpoint from resume_from_checkpoint. I chose this policy because resume_from_checkpoint is more explicit request from user.

**Comment by williamFalcon** [comment]
@dreamgonfly looks great. Merging this. We need to add docs for it though
